File: 02/2.hs
main = do
    text <- readFile "input.txt"
    let input = map parseLine $ lines text
    let result = run input
    --putStrLn $ show $ take 5 input
    putStrLn $ show result

type Line = ((Int,Int),Char,String)

parseLine :: String -> Line
parseLine xs = ((read mi, read ma), ch, pw)
    where [limits, (ch:_), pw] = words xs
          (mi,(_:ma)) = span (/='-') limits

run :: [Line] -> Int
run = length . filter p
    where p ((mi,ma), ch, pw) =
            let c1 = pw !! (mi - 1)
                c2 = pw !! (ma - 1)
            in (c1 == ch) /= (c2 == ch)

File: 02/2.js
const input = require("fs").readFileSync(process.argv[2] || "input.txt").toString().split("\n")
const lines = input.map(line => {
    let [limits, ch, password] = line.split(" ")
    const [min,max] = limits.split("-").map(Number).map(n => n-1)
    ch = ch[0]
    const valid = password[min] == ch ^ password[max] == ch
    return [valid, limits, ch, password]
})

console.log(lines.filter(l => l[0]).length)
